pkgs/os-specific/linux/bpftrace/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenv, fetchFromGitHub
+, cmake, pkgconfig, flex, bison
+, llvmPackages, kernel, linuxHeaders, elfutils, libelf, bcc
+}:
+
+stdenv.mkDerivation rec {
+  name = "bpftrace-${version}";
+  version = "git-2018-10-16";
+
+  src = fetchFromGitHub {
+    owner = "iovisor";
+    repo = "bpftrace";
+    rev = "51500dfbe2529823a261af2ab2879d3fb7cd8bf9";
+    sha256 = "0ajz5ddffp4c7c30bsxxis50wl1rk729bhdx7vrd41b9fgrwc6df";
+  };
+
+  # bpftrace requires an unreleased version of bcc, added to the cmake
+  # build as an ExternalProject.
+  # https://github.com/iovisor/bpftrace/issues/184
+  bccSrc = fetchFromGitHub {
+    owner = "iovisor";
+    repo = "bcc";
+    rev = "09f9d3c8dd016ee10dc8b57057763dc25b0d2208";
+    sha256 = "0axmppjg3s0835bmn5b2q3rli7l61ldgw7hh0922nn2zb2ghclbv";
+  };
+
+  buildInputs = [
+    llvmPackages.llvm llvmPackages.clang-unwrapped kernel
+    elfutils libelf bccSrc
+  ];
+
+  nativeBuildInputs = [ cmake pkgconfig flex bison ]
+    # libelf is incompatible with elfutils-libelf
+    ++ stdenv.lib.filter (x: x != libelf) kernel.moduleBuildDependencies;
+
+  patches = [
+    ./bcc-source.patch
+    # https://github.com/iovisor/bpftrace/issues/184
+    ./disable-gtests.patch
+    # https://github.com/iovisor/bpftrace/pull/183
```

Comment:
This could be ```fetchpatch { url = https://github.com/iovisor/bpftrace/pull/183.patch; sha256 = ...; }```

Reply:
That looks like a good idea but then if I make any review fixes to that PR then it will change the hash.

How about let's let this PR stew for a few days. Maybe that PR will get merged by then and then I can remove the patch altogether.

Reply:
OK, that PR is now merged, so I have updated the sources and removed the patch.